src/box/authentication.c

```diff
@@ -5,21 +5,48 @@
  */
 #include "authentication.h"
 
+#include <assert.h>
 #include <stdbool.h>
 #include <stdint.h>
 #include <stddef.h>
 #include <string.h>
 
+#include "assoc.h"
+#include "auth_chap_sha1.h"
 #include "base64.h"
 #include "diag.h"
 #include "errcode.h"
 #include "error.h"
+#include "fiber.h"
 #include "msgpuck.h"
 #include "scramble.h"
 #include "session.h"
+#include "small/region.h"
+#include "tt_static.h"
 #include "user.h"
 #include "user_def.h"
 
+const struct auth_method *AUTH_METHOD_DEFAULT;
+
+/** Map of all registered authentication methods: name -> auth_method. */
+static struct mh_strnptr_t *auth_methods = NULL;
+
+bool
+authenticate_password(const struct authenticator *auth,
+		      const char *password, int password_len)
+{
+	const char salt[SCRAMBLE_SIZE] = {0, };
```

Comment:
Removed the comma and added a comment explaining that any salt would do:

<details><summary>diff</summary>

```diff
diff --git a/src/box/authentication.c b/src/box/authentication.c
index 2fe192a9fcc2..454b026c8054 100644
--- a/src/box/authentication.c
+++ b/src/box/authentication.c
@@ -35,7 +35,13 @@ bool
 authenticate_password(const struct authenticator *auth,
 		      const char *password, int password_len)
 {
-	const char salt[SCRAMBLE_SIZE] = {0, };
+	/*
+	 * We don't really need to zero the salt here, because any salt would
+	 * do as long as we use the same salt in auth_request_prepare and
+	 * authenticate_request. We zero it solely to avoid address sanitizer
+	 * complaints about usage of uninitialized memory.
+	 */
+	const char salt[SCRAMBLE_SIZE] = {0};
 	struct region *region = &fiber()->gc;
 	size_t region_svp = region_used(region);
 	const char *auth_request, *auth_request_end;
```

</details>